### PR TITLE
[BugFix] fix parse number format json string behavior

### LINKED ISSUE #14688: Fix parse_json and cast string as json behavior
### Steps to reproduce the behavior (Required)
1. CREATE TABLE '...'
2. INSERT INTO '....'
3. SELECT  '....'

### StarRocks version (Required)
 - You can get the StarRocks version by executing SQL `select current_version()`

### PR DESCRIPTION
**Pull request by murphyatwork**
## What type of PR is this：
- [x] BugFix
- [ ] Feature
- [ ] Enhancement
- [ ] Refactor
- [ ] UT
- [ ] Doc
- [ ] Tool

## Which issues of this PR fixes ：

Fixes #14688

## Checklist:

- [ ] I have added test cases for my bug fix or my new feature
- [ ] This pr will affect users' behaviors
- [ ] This pr needs user documentation (for new or modified features or behaviors)
  - [ ] I have added documentation for my new feature or new function

## Bugfix cherry-pick branch check:
- [x] I have checked the version labels which the pr will be auto backported to target branch
  - [x] 2.5
  - [x] 2.4
  - [x] 2.3
  - [x] 2.2

### CODE REVIEW

File: be/src/util/json.cpp [hunk 1]
```diff
@@ -60,20 +60,15 @@ StatusOr<JsonValue> JsonValue::parse_json_or_string(const Slice& src) {
...
-        }
-        auto b = vpack::Parser::fromJson(src.get_data(), src.get_size());
-        out->assign(*b);
-    } catch (const vpack::Exception& e) {
-        return fromVPackException(e);
-    }
-    return Status::OK();
+    ASSIGN_OR_RETURN(auto json_value, parse_json_or_string(src));
+    *out = std::move(json_value);
+    return {};
```
Comment (jaogoy) [inline be/src/util/json.cpp:65]: why not directly use Status::OK();

**Comment by mofeiatwork** [comment]
run starrocks_admit_test

File: be/src/util/json.cpp (hunk 1 above)
Comment (mofeiatwork) [reply be/src/util/json.cpp:65]: It's shorter

**Comment by mofeiatwork** [comment]
run starrocks_admit_test

**Comment by mofeiatwork** [comment]
run starrocks_admit_test

**Review by satanson — APPROVED** [review]

**Review by kangkaisen — APPROVED** [review]

**Comment by wanpengfei-git** [comment]
@Mergifyio backport branch-2.5

**Comment by wanpengfei-git** [comment]
@Mergifyio backport branch-2.4

**Comment by wanpengfei-git** [comment]
run starrocks_admit_test

**Comment by wanpengfei-git** [comment]
@Mergifyio backport branch-2.2

**Pull request merged**